pkg/controller/certificaterequest/renew_certificate.go

```diff
@@ -59,6 +60,9 @@ func (r *ReconcileCertificateRequest) ShouldRenewOrReIssue(reqLogger logr.Logger
 		timeDiff := notAfter.Sub(currentTime)
 		daysCertificateValidFor := int(timeDiff.Hours() / 24)
 		shouldRenew := daysCertificateValidFor <= renewBeforeDays
+		certSerialNum := certificate.SerialNumber
+
+		localmetrics.UpdateCertExpiryDateMetric(cr.Spec.WebConsoleURL, certSerialNum, daysCertificateValidFor)
```

Comment:
The dimensions on this metric don't make much sense. You should log following dimensions for each metrics

- cr.Name
- cr.Spec.ACMEDNSDomain
- cert_expiry_month
- cert_expiry_day
- cert_expiry_year
- cert_expiring_in_days

Webconsole URL and Serial number are not needed.

Reply:
I disagree. Labels should only be used to categorize metrics, not have data in them.

Reply:
The dimensions used here currently don't help in any way with the metrics.

Reply:
They categorize the metric by signifying which cert/cluster is expiring.

Reply:
@fahlmant Yes and the dimensions added in the initial code do not allow for that. We do not want console url. We identify cluster using ACMEBaseDomain.

Reply:
Okay, by why cert expiry year, day, month and expiring in days? That will create a new metric for every day

Reply:
OK I have updated the dimension.

Reply:
LGTM :+1: